components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -1197,4 +1195,39 @@ public static String validateTenantDomain(String tenantDomain) throws UserStoreE
         return null;
     }
 
+    /**
+     * build the error response
+     * @param status
+     * @param message
+     * @return decoded response
+     * @throws org.wso2.carbon.identity.base.IdentityException
+     */
+    public static String buildErrorResponse(String status, String message)
+            throws IdentityException, IOException {
+        ErrorResponseBuilder respBuilder = new ErrorResponseBuilder();
+        List<String> statusCodeList = new ArrayList<String>();
+        statusCodeList.add(status);
+        Response response = respBuilder.buildResponse(null, statusCodeList, message);
+        String resp = SAMLSSOUtil.marshall(response);
+
+        return compressResponse(resp);
+
+    }
+
+    /**
+     * Compresses the response String
+     *
+     * @param response
+     * @return
+     * @throws IOException
+     */
+    public static String compressResponse(String response) throws IOException {
+        Deflater deflater = new Deflater(Deflater.DEFLATED, true);
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        DeflaterOutputStream deflaterOutputStream = new DeflaterOutputStream(byteArrayOutputStream, deflater);
+        deflaterOutputStream.write(response.getBytes());
+        deflaterOutputStream.close();
```

Comment:
Make it a habit that you always close streams or similar resources within a finally block. For example, imagine an exception is thrown while invoking the "write" method. Then the deflaterOutputStream will be left leading to a possible cause for OOM.